Fix cluster performance and reliability issues

## Description

During #25883, the cluster performance and reliability tests failed for different reasons. This issue aims at finding solutions to them.

### Performance tests

The `USS(KB)`, `CPU(%)` and `FD` values were higher than expected.

|value	|threshold	|stat	|column	|node	|file	|phase|
|-------|-----------|-----|-------|-----|-----|-----|
|1886.153523625221	|850	|reg_cof	|USS(KB)	|master|	wazuh-clusterd|	stable_phase|
|0.1599661344944362	|0.12	|reg_cof	|CPU(%)		|master|	wazuh-clusterd|	stable_phase|
|0.9479922593130139	|0.6	|reg_cof	|FD |master|	wazuh-clusterd |	stable_phase| 

### Reliability tests

#### test_cluster_error_logs

The test failed because the `cluster.log` file contains errors.

```console
2024/09/24 04:02:21 ERROR: [Local 62981] [Main] Internal error processing request 'b'sendsync'': Error 3023 - Worker node is not connected to master
2024/09/24 04:02:22 ERROR: [Local 952583] [Main] Internal error processing request 'b'sendsync'': Error 3023 - Worker node is not connected to master
2024/09/24 04:02:23 ERROR: [Local 197605] [Main] Internal error processing request 'b'sendsync'': Error 3023 - Worker node is not connected to master
2024/09/24 04:02:24 ERROR: [Local 71314] [Main] Internal error processing request 'b'sendsync'': Error 3023 - Worker node is not connected to master
```

#### test_check_logs_order_master

The test fails because of a `KeyError`.

```console
>                           incorrect_order[name].append({'node': name, 'log_type': log_tag,
                                                    'expected_logs': [log.tag for log in
                                                                      tree_info['tree'].children(tree_info['node'])],
                                                    'found_log': full_log})
E                           KeyError: 'Master'

wazuh-qa/tests/reliability/test_cluster/test_cluster_logs/test_cluster_master_logs_order/test_cluster_master_logs_order.py:93: KeyError
```

#### test_cluster_sync

The test fails because there's no integrity sync logs in the worker 25.

```console
if not sync_logs:
>                   pytest.fail(f'No integrity sync logs found in {node_name.search(log_file)[1]}')
E                   Failed: No integrity sync logs found in worker_25

wazuh-qa/tests/reliability/test_cluster/test_cluster_logs/test_cluster_sync/test_cluster_sync.py:45: Failed
```

#### test_check_logs_order_workers

The test fails because the order of the logs is not correct.

```console
E           worker_24
E            - Log type: Integrity check
E              Expected logs: ['Permission to synchronize granted.*']
E              Found log: 2024/09/24 03:35:34 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B732_manager_24] [Integrity check] Starting.
E           
E            - Log type: Agent-info sync
E              Expected logs: ['Permission to synchronize granted.*']
E              Found log: 2024/09/24 03:35:35 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B732_manager_24] [Agent-info sync] Starting.
E           
E            - Log type: Agent-groups recv
E              Expected logs: ['.*chunks updated in wazuh-db in.*']
E              Found log: 2024/09/24 03:35:44 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B732_manager_24] [Agent-groups recv] Finished in 0.014s. Updated 1 chunks.
```

Artifacts: [artifacts.zip](https://github.com/user-attachments/files/17131588/artifacts.zip)

## Update
### Key Error in `test_check_logs_order_master `

The error is caused by trying to access a key that doesn't exist in the dictionary. This dictionary starts without any values, so I added some validation to handle this case:

```python
                        if name not in incorrect_order:
                            incorrect_order[name] = []

                        data = {'node': name, 'log_type': log_tag,
                                                'expected_logs': [log.tag for log in
                                                                  tree_info['tree'].children(tree_info['node'])],
                                                'found_log': full_log}
                        incorrect_order[name].append(data)
                        pytest.fail(f"[{data['node']}]"
                                    f"\n - Log type: {data['log_type']}"
                                    f"\n - Expected logs: {data['expected_logs']}"
                                    f"\n - Found log: {data['found_log']}")
```

After these changes I ran the test, and the error disappeared:
> There are assert errors, but they are discussed 

<details>
<summary>Test Results</summary>

```console
 python3 -m pytest tests/reliability/test_cluster/test_cluster_logs/test_cluster_master_logs_order/test_cluster_master_logs_order.py --artifacts_path /Users/framos/Downloads/archive
=========================================================================== test session starts ============================================================================
platform darwin -- Python 3.9.6, pytest-7.1.2, pluggy-0.13.1
rootdir: /Users/framos/Documents/Wazuh/Repositories/wazuh-qa/tests, configfile: pytest.ini
plugins: testinfra-5.0.0, html-3.1.1, metadata-2.0.4
collected 1 item                                                                                                                                                           

tests/reliability/test_cluster/test_cluster_logs/test_cluster_master_logs_order/test_cluster_master_logs_order.py F                                                  [100%]

================================================================================= FAILURES =================================================================================
_______________________________________________________________________ test_check_logs_order_master _______________________________________________________________________

artifacts_path = '/Users/framos/Downloads/archive'

    def test_check_logs_order_master(artifacts_path):
        """Check that cluster logs appear in the expected order.
    
        Check that for each group of logs (agent-info, integrity-check, etc), each message
        appears in the order it should. If any log is duplicated, skipped, etc. the test will fail.
    
        Args:
            artifacts_path (str): Path where folders with cluster information can be found.
        """
        if not artifacts_path:
            pytest.fail('Parameter "--artifacts_path=<path>" is required.')
    
        cluster_log_files = os.path.join(artifacts_path, 'master', 'logs', 'cluster.log')
    
        if not os.path.exists(cluster_log_files):
            pytest.fail(f"No files found inside {artifacts_path}.")
    
        all_managers = {'Master': LogsOrder().logs_order}
        name = ''
    
        with open(cluster_log_files) as file:
            for line in file.readlines():
                result = logs_format.search(line)
                if result:
                    node_name = result.group(1)
                    if 'Worker' in node_name:
                        name = re.search('.*Worker (.*?)]', node_name).group(1)
                        if name not in all_managers:
                            all_managers[name] = LogsOrder().logs_order
                    elif 'Master' in node_name:
                        name = 'Master'
    
                    log_tag = result.group(2)
                    full_log = result.group(3)
    
                    if log_tag in all_managers[name]:
                        tree_info = all_managers[name][log_tag]
                        for child in tree_info['tree'].children(tree_info['node']):
                            if re.search(child.tag, full_log):
                                # Current node is updated so the tree points to the next expected log.
                                all_managers[name][log_tag]['node'] = child.identifier if \
                                    tree_info['tree'].children(child.identifier) else 'root'
                                break
                        else:
                            if name not in incorrect_order:
                                incorrect_order[name] = []
    
                            data = {'node': name, 'log_type': log_tag,
                                                    'expected_logs': [log.tag for log in
                                                                      tree_info['tree'].children(tree_info['node'])],
                                                    'found_log': full_log}
                            incorrect_order[name].append(data)
>                           pytest.fail(f"[{data['node']}]"
                                        f"\n - Log type: {data['log_type']}"
                                        f"\n - Expected logs: {data['expected_logs']}"
                                        f"\n - Found log: {data['found_log']}")
E                           Failed: [Master]
E                            - Log type: Local agent-groups
E                            - Expected logs: ['Obtained.*chunks of data in.*', 'No clients connected. Skipping.*']
E                            - Found log: Finished in 0.001s.

tests/reliability/test_cluster/test_cluster_logs/test_cluster_master_logs_order/test_cluster_master_logs_order.py:101: Failed
========================================================================= short test summary info ==========================================================================
FAILED tests/reliability/test_cluster/test_cluster_logs/test_cluster_master_logs_order/test_cluster_master_logs_order.py::test_check_logs_order_master - Failed: [Master]
============================================================================ 1 failed in 0.07s =============================================================================
```
</details>

### Assert error in `test_cluster_error_logs `

This test is in charge of checking if there is any error message in the log. The error that triggered this is the following:

```console
2024/09/24 04:00:17 ERROR: [Local Server] [Main] Could not connect to master. Trying again in 10 seconds.
2024/09/24 04:00:27 ERROR: [Local Server] [Main] Could not connect to master. Trying again in 10 seconds.
2024/09/24 04:00:37 ERROR: [Local Server] [Main] Could not connect to master. Trying again in 10 seconds.
2024/09/24 04:00:47 ERROR: [Local Server] [Main] Could not connect to master. Trying again in 10 seconds.
2024/09/24 04:00:57 ERROR: [Local Server] [Main] Could not connect to master. Trying again in 10 seconds.
2024/09/24 04:01:07 ERROR: [Local Server] [Main] Could not connect to master. Trying again in 10 seconds.
2024/09/24 04:01:17 ERROR: [Local Server] [Main] Could not connect to master. Trying again in 10 seconds.
2024/09/24 04:01:27 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B732_manager_1] [Main] Successfully connected to master.

....
2024/09/24 04:02:21 ERROR: [Local 62981] [Main] Internal error processing request 'b'sendsync'': Error 3023 - Worker node is not connected to master
2024/09/24 04:02:22 ERROR: [Local 952583] [Main] Internal error processing request 'b'sendsync'': Error 3023 - Worker node is not connected to master
2024/09/24 04:02:23 ERROR: [Local 197605] [Main] Internal error processing request 'b'sendsync'': Error 3023 - Worker node is not connected to master
2024/09/24 04:02:24 ERROR: [Local 71314] [Main] Internal error processing request 'b'sendsync'': Error 3023 - Worker node is not connected to master
```

This error is raised in this part of the code:

https://github.com/wazuh/wazuh/blob/8467a644df8ad3aa8b40836a8dc60595d7878ac2/framework/wazuh/core/cluster/local_server.py#L362-L402

Finally, checking the `cluster.log` of the `master`, this disconnection error where caused due to the cluster being restarted:

```console
2024/09/24 04:00:07 INFO: [Cluster] [Main] SIGNAL [(15)-(SIGTERM)] received. Exit...
2024/09/24 04:00:07 ERROR: [Cluster] [Main] Error while trying to terminate the process with ID 25871.
2024/09/24 04:01:26 INFO: [Local Server] [Main] Serving on /var/ossec/queue/cluster/c-internal.sock
2024/09/24 04:01:26 INFO: [Master] [Main] Serving on ('0.0.0.0', 1516)

......

2024/09/24 04:01:38 INFO: [Cluster] [Main] SIGNAL [(15)-(SIGTERM)] received. Exit...
2024/09/24 04:02:19 INFO: [Local Server] [Main] Serving on /var/ossec/queue/cluster/c-internal.sock
2024/09/24 04:02:19 INFO: [Master] [Main] Serving on ('0.0.0.0', 1516)
2024/09/24 04:02:19 INFO: [Master] [Local integrity] Starting.
2024/09/24 04:02:19 INFO: [Master] [Local agent-groups] Sleeping 30s before starting the agent-groups task, waiting for the workers connection.
2024/09/24 04:02:19 INFO: [Master] [Local integrity] Finished in 0.199s. Calculated metadata of 34 files.
2024/09/24 04:02:25 INFO: [Worker] [Main] Connection from ('172.31.2.29', 34526)
2024/09/24 04:02:25 INFO: [Worker] [Main] Connection from ('172.31.2.141', 50076)
2024/09/24 04:02:25 INFO: [Worker] [Main] Connection from ('172.31.4.171', 55320)
2024/09/24 04:02:25 INFO: [Worker] [Main] Connection from ('172.31.8.8', 60940)
2024/09/24 04:02:25 INFO: [Worker] [Main] Connection from ('172.31.14.40', 51888)
2024/09/24 04:02:25 INFO: [Worker] [Main] Connection from ('172.31.6.22', 44558)
2024/09/24 04:02:25 INFO: [Worker] [Main] Connection from ('172.31.11.153', 39502)
2024/09/24 04:02:25 INFO: [Worker] [Main] Connection from ('172.31.6.90', 33896)
2024/09/24 04:02:25 INFO: [Worker] [Main] Connection from ('172.31.15.13', 51594)
2024/09/24 04:02:25 INFO: [Worker] [Main] Connection from ('172.31.7.180', 34636)
2024/09/24 04:02:25 INFO: [Worker] [Main] Connection from ('172.31.14.110', 40292)
2024/09/24 04:02:26 INFO: [Worker] [Main] Connection from ('172.31.13.136', 50854)
2024/09/24 04:02:26 INFO: [Worker] [Main] Connection from ('172.31.8.63', 39256)
2024/09/24 04:02:26 INFO: [Worker] [Main] Connection from ('172.31.3.54', 52326)
2024/09/24 04:02:26 INFO: [Worker] [Main] Connection from ('172.31.1.101', 53328)
2024/09/24 04:02:26 INFO: [Worker] [Main] Connection from ('172.31.3.202', 50178)
2024/09/24 04:02:26 INFO: [Worker] [Main] Connection from ('172.31.1.212', 49392)
2024/09/24 04:02:27 INFO: [Worker] [Main] Connection from ('172.31.2.174', 41102)
2024/09/24 04:02:27 INFO: [Worker] [Main] Connection from ('172.31.6.223', 47250)
2024/09/24 04:02:27 INFO: [Worker] [Main] Connection from ('172.31.4.223', 50422)
2024/09/24 04:02:27 INFO: [Worker] [Main] Connection from ('172.31.13.49', 50058)
2024/09/24 04:02:27 INFO: [Worker] [Main] Connection from ('172.31.4.0', 57462)
2024/09/24 04:02:27 INFO: [Worker] [Main] Connection from ('172.31.4.44', 59200)
2024/09/24 04:02:27 INFO: [Worker] [Main] Connection from ('172.31.2.162', 35388)
2024/09/24 04:02:27 INFO: [Worker] [Main] Connection from ('172.31.14.192', 48322)
```

--- a/tests/reliability/test_cluster/test_cluster_logs/test_cluster_master_logs_order/test_cluster_master_logs_order.py
+++ b/tests/reliability/test_cluster/test_cluster_logs/test_cluster_master_logs_order/test_cluster_master_logs_order.py
@@ -90,11 +90,15 @@ def test_check_logs_order_master(artifacts_path):
                                 tree_info['tree'].children(child.identifier) else 'root'
                             break
                     else:
-                        incorrect_order[name].append({'node': name, 'log_type': log_tag,
+                        if name not in incorrect_order:
+                            incorrect_order[name] = []
+
+                        data = {'node': name, 'log_type': log_tag,
                                                 'expected_logs': [log.tag for log in
                                                                   tree_info['tree'].children(tree_info['node'])],
-                                                'found_log': full_log})
-                        pytest.fail(f"[{incorrect_order[name]['node']}]"
-                                    f"\n - Log type: {incorrect_order[name]['log_type']}"
-                                    f"\n - Expected logs: {incorrect_order[name]['expected_logs']}"
-                                    f"\n - Found log: {incorrect_order[name]['found_log']}")
+                                                'found_log': full_log}
+                        incorrect_order[name].append(data)
+                        pytest.fail(f"[{data['node']}]"
+                                    f"\n - Log type: {data['log_type']}"
+                                    f"\n - Expected logs: {data['expected_logs']}"
+                                    f"\n - Found log: {data['found_log']}")

